### PR TITLE
Flow: Docs for discovery.kubernetes

### PR DESCRIPTION
**Pull request by captncraig**
First draft. 

- Http stuff copied from prometheus.scrape. Would be great to have a way to include a partial.
- Some of the selector stuff is copied and adapted from prometheus docs, or links out. Because some of it is a little tricky, and I'm not sure how much we want to duplicate, or re-state.

### CODE REVIEW

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+---
+aliases:
+- /docs/agent/latest/flow/reference/components/discovery.kubernetes
+title: discovery.kubernetes
+---
+
+# discovery.kubernetes
+
+`discovery.kubernetes` allows you to find scrape targets from kubernetes resources. It watches cluster state, and ensures targets are continually synced with what is currently running in your cluster.
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:9]:
```suggestion
`discovery.kubernetes` allows you to find scrape targets from Kubernetes resources. It watches cluster state, and ensures targets are continually synced with what is currently running in your cluster.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+Internally, this component uses kubernetes service discovery from prometheus. [The documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) on their site offers further details on the metadata labels applied to targets and other usage considerations.
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:11]:
```suggestion
Internally, this component uses Kubernetes Service Discovery from Prometheus. [The documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) on their site offers further details on the metadata labels applied to targets and other usage considerations.
```

I'd prefer avoiding linking out to Prometheus documentation to avoid issues where we become out of sync if upstream updates before we have a chance to support something. Let's reproduce what we need to explain to users here instead of linking out.

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+If you supply no connection information, this component will default to using an in-cluster config. Otherwise, you can specify a kubeconfig file, or manual apiserver configuration.
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:13]: Use present tense as much as possible:

https://developers.google.com/style/tense?hl=en

```suggestion
If you supply no connection information, this component defaults to using an in-cluster config. Otherwise, you can specify a kubeconfig file, or manual apiserver configuration.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`api_server` | `string` | URL of kubernetes API server | | no
+`role` | `string` | Type of kubernetes resource to query. Must be one of `node`, `pod`, `service`, `endpoints`, `endpointslice` or `ingress` | | **yes**
+`kubeconfig_file` | `string` | Path of kubeconfig file to use for kubernetes connection | | no
+
+The following subblocks are supported:
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:47]:
```suggestion
The following sub-blocks are supported:
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+Optional namespace discovery. If omitted, all namespaces are used.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`own_namespace` | `bool`   | Include the namespace the agent is running in? | | no
+`names` | `[]string` | List of namespaces | | no
+
+### `selectors` block
+
+This contains optional label and field selectors to limit the discovery process to a subset of available resources.
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:66]:
```suggestion
The `selectors` block contains optional label and field selectors to limit the discovery process to a subset of resources.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+See [https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/)
+and [https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) to learn more about the possible
+filters that can be used.
+
+Note: When making decision about using field/label selector make sure that this
+is the best approach - it will prevent Prometheus from reusing single list/watch
+for all scrape configs. This might result in a bigger load on the Kubernetes API,
+because per each selector combination there will be additional LIST/WATCH. On the other hand,
+if you just want to monitor small subset of pods in large cluster it's recommended to use selectors.
+The decision, to use selectors or not depends on the particular situation.
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:76]: I find this paragraph a little hard to follow. Can you try to rephrase this less conversationally?

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`api_server` | `string` | URL of kubernetes API server | | no
+`role` | `string` | Type of kubernetes resource to query. Must be one of `node`, `pod`, `service`, `endpoints`, `endpointslice` or `ingress` | | **yes**
+`kubeconfig_file` | `string` | Path of kubeconfig file to use for kubernetes connection | | no
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:45]:
```suggestion
`kubeconfig_file` | `string` | Path of kubeconfig file to use for Kubernetes connection | | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`api_server` | `string` | URL of kubernetes API server | | no
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:43]:
```suggestion
`api_server` | `string` | URL of Kubernetes API server | | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+`role` | `string` | Type of kubernetes resource to query. Must be one of `node`, `pod`, `service`, `endpoints`, `endpointslice` or `ingress` | | **yes**
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:44]:
```suggestion
`role` | `string` | Type of Kubrnetes resource to query. Must be one of `node`, `pod`, `service`, `endpoints`, `endpointslice` or `ingress` | | **yes**
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+`kubeconfig_file` | `string` | Path of kubeconfig file to use for kubernetes connection | | no
+
+The following subblocks are supported:
+
+Name | Description | Required
+---- | ----------- | --------
+[`namespaces`](#namespaces-block) | Information about which kubernetes namespaces to search. | no
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:51]:
```suggestion
[`namespaces`](#namespaces-block) | Information about which Kubernetes namespaces to search. | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+[`selectors`](#selectors-block) | Selectors to limit objects selected | no
+[`http_client_config`](#http_client_config-block) | HTTP client configuration for kubernetes requests | no
+
+### `namespaces` block
+
+Optional namespace discovery. If omitted, all namespaces are used.
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:57]:
```suggestion
The `namespaces` block limits the namespaces to discover resources in. If omitted, all namespaces are searched.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+The decision, to use selectors or not depends on the particular situation.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`role` | `string`   | role of the selector | | yes
+`label`| `string`   | label selector string | | no
+`field` | `string`   | field selector string | | no
+
+### `http_client_config` block
+
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:85]:
```suggestion
### `http_client_config` block

The `http_client_config` block configures settings used to connect to the Kubernetes API server.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+`enable_http_2`            | `bool`     | Whether the scraper should use HTTP2. | `true` | no
+
+The following sub-blocks are supported for `http_client_config`:
+
+Name | Description | Required
+---- | ----------- | --------
+[`basic_auth`](#basic_auth-block) | Configure basic_auth for authenticating against targets | no
+[`authorization`](#authorization-block) | Configure generic authorization against targets | no
+[`oauth2`](#oauth2-block) | Configure OAuth2 for authenticating against targets | no
+[`tls_config`](#tls_config-block) | Configure TLS settings for connecting to targets | no
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:101]:
```suggestion
[`basic_auth`](#basic_auth-block) | Configure basic_auth for authenticating against Kubernetes | no
[`authorization`](#authorization-block) | Configure generic authorization against Kubernetes | no
[`oauth2`](#oauth2-block) | Configure OAuth2 for authenticating against Kubernetes | no
[`tls_config`](#tls_config-block) | Configure TLS settings for connecting to Kubernetes | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+
+#### `tls_config` block
+
+Name                              | Type       | Description                                | Default | Required
+--------------------------------- | ---------- | ------------------------------------------ | ------- | --------
+`tls_config_ca_file`              | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_cert_file`            | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_key_file`             | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_server_name`          | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_insecure_skip_verify` | `bool`     | Configuration options for TLS connections. |         | no
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:141]: Can you copy the documentation for basic_auth, authorization, oauth2, and tls_config from [prometheus.remote_write](https://github.com/grafana/agent/blob/main/docs/flow/reference/components/prometheus.remote_write.md#basic_auth-block) instead? It generally provides more information.

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+targets | list(map(string)) | The set of targets discovered from the kubernetes api
```
Comment (rfratto) [inline docs/flow/reference/components/discovery.kubernetes.md:149]:
```suggestion
targets | list(map(string)) | The set of targets discovered from the Kubernetes API
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+aliases:
+- /docs/agent/latest/flow/reference/components/discovery.kubernetes
+title: discovery.kubernetes
+---
+
+# discovery.kubernetes
+
+`discovery.kubernetes` allows you to find scrape targets from Kubernetes resources. It watches cluster state, and ensures targets are continually synced with what is currently running in your cluster.
+
+Internally, this component uses Kubernetes Service Discovery from Prometheus. [The documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) on their site offers further details on the metadata labels applied to targets and other usage considerations.
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:11]:
```suggestion
Internally, this component uses Kubernetes Service Discovery from Prometheus. [The documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) on the Prometheus site offers further details on the metadata labels applied to targets and other usage considerations.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+
+If you supply no connection information, this component defaults to using an in-cluster config. Otherwise, you can specify a kubeconfig file, or manual apiserver configuration.
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:13]:
```suggestion
If you supply no connection information, this component defaults to an in-cluster config. Otherwise, you can specify a kubeconfig file or a manual apiserver configuration.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+
+## Example
+
+Simple in-cluster discovery of all pods:
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:17]:
```suggestion
This example shows a simple in-cluster discovery of all pods:
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 19]
```diff
@@ -0,0 +1,164 @@
...
+
+```river
+discovery.kubernetes "k8s_pods" {
+  role = "pod"
+}
+```
+
+Specific namespace only with kubeconfig file:
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:25]:
```suggestion
You only need to specify a namespace for kubeconfig files:
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 20]
```diff
@@ -0,0 +1,164 @@
...
+
+```river
+discovery.kubernetes "k8s_pods" {
+  role = "pod"
+  kubeconfig_file = "/path/to/kubeconfig"
+  namepsaces {
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:31]:
```suggestion
  namespaces {
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+`api_server` | `string` | URL of kubernetes API server | | no
+`role` | `string` | Type of kubernetes resource to query. Must be one of `node`, `pod`, `service`, `endpoints`, `endpointslice` or `ingress` | | **yes**
+`kubeconfig_file` | `string` | Path of kubeconfig file to use for kubernetes connection | | no
+
+The following sub-blocks are supported:
+
+Name | Description | Required
+---- | ----------- | --------
+[`namespaces`](#namespaces-block) | Information about which kubernetes namespaces to search. | no
+[`selectors`](#selectors-block) | Selectors to limit objects selected | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:52]:
```suggestion
[`selectors`](#selectors-block) | Selectors to limit objects selected. | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
+[`http_client_config`](#http_client_config-block) | HTTP client configuration for kubernetes requests | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:53]:
```suggestion
[`http_client_config`](#http_client_config-block) | HTTP client configuration for Kubernetes requests. | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 23]
```diff
@@ -0,0 +1,164 @@
...
+
+### `namespaces` block
+
+The `namespaces` block limits the namespaces to discover resources in. If omitted, all namespaces are searched.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`own_namespace` | `bool`   | Include the namespace the agent is running in? | | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:61]:
```suggestion
`own_namespace` | `bool`   | Include the namespace the agent is running in. | | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 24]
```diff
@@ -0,0 +1,164 @@
...
+`names` | `[]string` | List of namespaces | | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:62]:
```suggestion
`names` | `[]string` | List of namespaces. | | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 25]
```diff
@@ -0,0 +1,164 @@
...
+
+### `selectors` block
+
+The `selectors` block contains optional label and field selectors to limit the discovery process to a subset of resources.
+See [https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/)
+and [https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) to learn more about the possible
+filters that can be used.
+
+Note: When making decision about using field/label selector make sure that this
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:71]:
```suggestion
Note: Using field/label selectors 
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 26]
```diff
@@ -0,0 +1,164 @@
...
+is the best approach - it will prevent Prometheus from reusing single list/watch
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:72]:
```suggestion
prevents Prometheus from reusing a single `list/watch`
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 27]
```diff
@@ -0,0 +1,164 @@
...
+for all scrape configs. This might result in a bigger load on the Kubernetes API,
+because per each selector combination there will be additional LIST/WATCH. On the other hand,
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:74]:
```suggestion
because there will be an additional `list/watch' for each selector combination. 
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 28]
```diff
@@ -0,0 +1,164 @@
...
+if you just want to monitor small subset of pods in large cluster it's recommended to use selectors.
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:75]:
```suggestion
If you just want to monitor a small subset of pods in a large cluster, it's recommended that you use selectors.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 29]
```diff
@@ -0,0 +1,164 @@
...
+The decision, to use selectors or not depends on the particular situation.
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:76]:
```suggestion
The decision to use selectors or not depends on the particular situation.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 30]
```diff
@@ -0,0 +1,164 @@
...
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`role` | `string`   | role of the selector | | yes
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:80]:
```suggestion
`role` | `string`   | Role of the selector. | | yes
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 31]
```diff
@@ -0,0 +1,164 @@
...
+`label`| `string`   | label selector string | | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:81]:
```suggestion
`label`| `string`   | Label selector string. | | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 32]
```diff
@@ -0,0 +1,164 @@
...
+`field` | `string`   | field selector string | | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:82]:
```suggestion
`field` | `string`   | Field selector string. | | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 33]
```diff
@@ -0,0 +1,164 @@
...
+
+### `http_client_config` block
+
+The `http_client_config` block configures settings used to connect to the Kubernetes API server.
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:86]:
```suggestion
The `http_client_config` block configures settings used to connect to the Kubernetes API server.

```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 34]
```diff
@@ -0,0 +1,164 @@
...
+`bearer_token_file`        | `string`   | Use to set up the Bearer Token file. | | no
+`proxy_url`                | `string`   | Use to set up a proxy URL. | | no
+`follow_redirects`         | `bool`     | Whether the scraper should follow redirects. | `true` | no
+`enable_http_2`            | `bool`     | Whether the scraper should use HTTP2. | `true` | no
+
+The following sub-blocks are supported for `http_client_config`:
+
+Name | Description | Required
+---- | ----------- | --------
+[`basic_auth`](#basic_auth-block) | Configure basic_auth for authenticating against Kubernetes | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:99]:
```suggestion
[`basic_auth`](#basic_auth-block) | Configure basic_auth for authenticating against Kubernetes. | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 35]
```diff
@@ -0,0 +1,164 @@
...
+[`authorization`](#authorization-block) | Configure generic authorization against Kubernetes | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:100]:
```suggestion
[`authorization`](#authorization-block) | Configure generic authorization against Kubernetes. | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 36]
```diff
@@ -0,0 +1,164 @@
...
+[`oauth2`](#oauth2-block) | Configure OAuth2 for authenticating against Kubernetes | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:101]:
```suggestion
[`oauth2`](#oauth2-block) | Configure OAuth2 for authenticating against Kubernetes. | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 37]
```diff
@@ -0,0 +1,164 @@
...
+[`tls_config`](#tls_config-block) | Configure TLS settings for connecting to Kubernetes | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:102]:
```suggestion
[`tls_config`](#tls_config-block) | Configure TLS settings for connecting to Kubernetes. | no
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 38]
```diff
@@ -0,0 +1,164 @@
...
+
+#### `basic_auth` block
+
+Name          | Type     | Description                                     | Default | Required
+------------- | -------- | ----------------------------------------------- | ------- | -------
+`username`      | `string`   | Setup of Basic HTTP authentication credentials. |         | no
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:108]: I think we should describe the fields rather than repeat "Setup of ... credentials."  Maybe just remove the Description column if there's nothing useful to say for all of these sub-blocks...

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 39]
```diff
@@ -0,0 +1,164 @@
...
+`tls_config_server_name`          | `string`   | Configuration options for TLS connections. |         | no
+`tls_config_insecure_skip_verify` | `bool`     | Configuration options for TLS connections. |         | no
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+targets | list(map(string)) | The set of targets discovered from the Kubernetes API
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:150]:
```suggestion
`targets` | `list(map(string))` | The set of targets discovered from the Kubernetes API.
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 40]
```diff
@@ -0,0 +1,164 @@
...
+
+## Component health
+
+`discovery.kubernetes` will only be reported as unhealthy when given an invalid
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:154]:
```suggestion
`discovery.kubernetes` is only reported as unhealthy when given an invalid
```

File: docs/flow/reference/components/discovery.kubernetes.md [hunk 41]
```diff
@@ -0,0 +1,164 @@
...
+configuration. In those cases, exported fields will be kept at their last
```
Comment (karengermond) [inline docs/flow/reference/components/discovery.kubernetes.md:155]:
```suggestion
configuration. In those cases, exported fields retain their last
```

**Review by karengermond — APPROVED** [review]
LGTM!
